src/main/java/taxi/dao/jdbc/CarJdbcDao.java

```diff
@@ -0,0 +1,199 @@
+package taxi.dao.jdbc;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import taxi.dao.CarDao;
+import taxi.lib.Dao;
+import taxi.model.Car;
+import taxi.model.Driver;
+import taxi.model.Manufacturer;
+import taxi.services.util.ConnectionUtil;
+
+@Dao
+public class CarJdbcDao implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars (model, manufacturer_id) VALUES (?,?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query,
+                        PreparedStatement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't create car" + car.getModel(), e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getCarQuery = "SELECT * FROM cars WHERE id = ? AND deleted = false";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(getCarQuery)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                car = getCarFromResult(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Cant get car by id " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversFromCar(car.getId()));
+        }
+        return Optional.empty();
```

Comment:
do we return Optional.empty() always from this method?

Reply:
I was thinking, that we will never reach this line, if everything else works correctly

Reply:
our car may be still null, or it may be initialised if everything is ok
there is no return from our method until this point